tests/components/yandex_transport/test_yandex_transport_sensor.py

```diff
@@ -0,0 +1,138 @@
+    """Check transport filtering by routes list."""
+    closer_time = None
+    if filter_routes is None:
+        filter_routes = []
+    attrs = {}
+
+    stop_metadata = reply["data"]["properties"]["StopMetaData"]
+
+    stop_name = reply["data"]["properties"]["name"]
+    transport_list = stop_metadata["Transport"]
+    for transport in transport_list:
+        route = transport["name"]
+        if filter_routes and route not in filter_routes:
+            # skip unnecessary route info
+            continue
+        if "Events" in transport["BriefSchedule"]:
+            for event in transport["BriefSchedule"]["Events"]:
+                if "Estimated" in event:
+                    posix_time_next = int(event["Estimated"]["value"])
+                    if closer_time is None or closer_time > posix_time_next:
+                        closer_time = posix_time_next
+                    if route not in attrs:
+                        attrs[route] = []
+                    attrs[route].append(event["Estimated"]["text"])
+    attrs["stop_name"] = stop_name
+    attrs[ATTR_ATTRIBUTION] = "Data provided by maps.yandex.ru"
+    if closer_time is None:
+        state = None
+    else:
+        state = dt_util.utc_from_timestamp(closer_time).isoformat(timespec="seconds")
+    return attrs, state
+
+
+async def assert_setup_sensor(hass, config, count=1):
+    """Set up the sensor and assert it's been created."""
+    with assert_setup_component(count):
+        assert await async_setup_component(hass, sensor.DOMAIN, config)
+
+
+async def test_setup_platform_valid_config(hass, mock_requester):
+    """Test that sensor is set up properly with valid config."""
+    await assert_setup_sensor(hass, TEST_CONFIG)
+
+
+async def test_setup_platform_invalid_config(hass, mock_requester):
+    """Check an invalid configuration."""
+    await assert_setup_sensor(
+        hass, {"sensor": {"platform": "yandex_transport", "stopid": 1234}}, count=0
+    )
+
+
+async def test_name(hass, mock_requester):
+    """Return the name if set in the configuration."""
+    await assert_setup_sensor(hass, TEST_CONFIG)
+    state = hass.states.get("sensor.test_name")
+    assert state.name == TEST_CONFIG["sensor"][CONF_NAME]
+
+
+async def test_state(hass, mock_requester):
+    """Return the contents of _state."""
+    await assert_setup_sensor(hass, TEST_CONFIG)
+    state = hass.states.get("sensor.test_name")
+    assert state.state == dt_util.utc_from_timestamp(1568659253).isoformat(
+        timespec="seconds"
+    )
+
+
+async def test_filtered_attributes(hass, mock_requester):
+    """Return the contents of attributes."""
+    await assert_setup_sensor(hass, TEST_CONFIG)
+    state = hass.states.get("sensor.test_name")
+
+    true_attrs, true_state = true_filter(REPLY, filter_routes=ROUTES)
+    assert state.state == true_state
+    result_key_len = len(true_attrs)
+    i = 0
+    assert i != result_key_len
+    for key in true_attrs:
```

Comment:
We can simplify this I think.
```py
assert dict(state.attributes) == true_attrs
```

Reply:
if i'm not mistaken, state.attributes contain more when we need

Reply:
Ok, how about:
```py
state_attrs = {key: state.attributes[key] for key in true_attrs}
assert state_attrs == true_attrs
```